src/main/java/com/amazon/opendistroforelasticsearch/security/OpenDistroSecurityPlugin.java

```diff
@@ -255,7 +252,7 @@ public OpenDistroSecurityPlugin(final Settings settings, final Path configPath)
             return;
         }
         
-        sslOnly = isSslOnlyMode(settings);
+        sslOnly = openDistroSSLConfig.isSslOnlyMode();
```

Comment:
I think it is not necessary to have `sslOnly` both in `OpenDistroSSLConfig` and `OpenDistroSecurityPlugin`, especially that `OpenDistroSecurityPlugin` directly references `OpenDistroSSLConfig`.

Reply:
currently in ```OpenDistroSecurityPlugin``` constructor the sslOnly value is being set to false if ```opendistro_security.disabled``` is set to false. 
If we want to replace all usage of ```sslOnly``` in ```OpenDistroSecurityPlugin``` then we would need to add ```isDisabled()``` check also in all those places. 
i.e. all usages of ```sslOnly``` in OpenDistroSecurityPlugin would need to be replaced with ```(!disabled && openDistroSSLConfig.isSslOnlyMode())```

Please let me know if I should proceed with those changes.

Reply:
I don't see a need to introduce `isDisabled()` check. When `opendistro_security.disabled` is `true`, `sslOnly` is initialized to `false`. It is still necessary to initialize `OpenDistroSSLConfig` when `opendistro_security.disabled` is `true` and set `sslOnly` to `false`. If you have a different approach in mind, let's discuss.

Reply:
agree, I have updated the code to use ```openDistroSSLConfig.isSslOnlyMode()``` instead of ```sslOnly``` member variable in  ```OpenDistroSecurityPlugin```